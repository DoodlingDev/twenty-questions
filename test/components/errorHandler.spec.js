/* eslint require-jsdoc: "off" */
import React from "react";
import ErrorHandler from "../../src/components/errorHandler.js";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const setupProps = {
  name: "test-error",
};

it("renders corrrectly", () => {
  const render = renderer.create(
    <ErrorHandler
      {...setupProps}
    />
  ).toJSON();
  expect(render).toMatchSnapshot();
});
