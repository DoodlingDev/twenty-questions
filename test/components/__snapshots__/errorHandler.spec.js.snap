// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when validation fails 1`] = `
<div
  className="q20_testError__errorHandler--errorData"
>
  <ul
    className="q20_testError__errorHandler--errorList"
  >
    <li
      className="q20_testError__errorHandler--errorMessage"
    >
      This validation failed!
    </li>
  </ul>
</div>
`;

exports[`renders correctly when validation passes w/ label 1`] = `
<div
  className="q20_testError__errorHandler--validData"
>
  <span
    className="q20_testError__errorHandler--validIndicator"
  >
    test Label ✔︎
  </span>
</div>
`;

exports[`renders correctly when validation passes w/o label 1`] = `
<div
  className="q20_testError__errorHandler--validData"
>
  <span
    className="q20_testError__errorHandler--validIndicator"
  >
     ✔︎
  </span>
</div>
`;

exports[`renders corrrectly before error exists 1`] = `
<div
  className="q20_testError__errorHandler"
/>
`;
